src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs

```diff
@@ -175,13 +175,18 @@ private void ValidateSecurityModelsMasking(IEnumerable<RegexPattern> patterns, I
 
                         foreach (string testExample in pattern.GenerateTruePositiveExamples())
                         {
-                            string secretValue = testExample;
+                            var detection = secretMasker.DetectSecrets(testExample).FirstOrDefault();
```

Comment:
>detection [](http://example.com/codeflow?start=32&length=9)

Here's a second way to obtain the literal secret. Run the detection and use the location data to get the secret value. 

This approach is much more what you'd expect in our API scenarios. I've left this inconsistent consciously. The other approach is more concise and expedient for tests.